core/src/main/java/hudson/model/StringParameterDefinition.java

```diff
@@ -60,14 +74,42 @@ public String getDefaultValue() {
         return defaultValue;
     }
 
+    /**
+     * 
+     * @return original or trimmed defaultValue (depending on trim)
+     * @since TODO
+     */
+    @Restricted(NoExternalUse.class)
+    public String getDefaultValue4Build() {
+        if (isTrim() && defaultValue != null) {
```

Comment:
I wonder whether this could be simplified with Util.fixNull(andTrim) or what they're called.